Peers Of New Configurations Appear Offline

I currently have 24 network configurations. If I add 3 more network configurations they display fine.

If I add a 4th network configuration the peers show as offline. The WireGuard tunnel operates fine however. The same happens for any subsequent network configurations that are added.

Restarting WGDashboard fixes the display issue and the peers of the 4th and subsequent network configurations show as online as expected.

The issue only started happening after having more than approximately 20 network configurations.

**To Reproduce**
1. Have 20+ network configurations.
2. Add 3 more network configurations. Peers display as online as expected.
3. Add a 4th, 5th, 6th... network configuration. Peers show as offline.
4. Restart WGDashboard. Peers now show as online as expected.
5. Repeats for every 4th+ configuration after each restart.

**OS Information:**
 - OS: Ubuntu 24.04.2 LTS
 - Python Version: 3.12.3

There's a lot of error files all containing the same (except for the times):

```
[2025-05-29 07:00:06 +0930] [1410634] [INFO] Starting gunicorn 23.0.0
[2025-05-29 07:00:06 +0930] [1410634] [ERROR] Connection in use: ('0.0.0.0', 10086)
[2025-05-29 07:00:06 +0930] [1410634] [ERROR] connection to ('0.0.0.0', 10086) failed: [Errno 98] Address>
[2025-05-29 07:00:07 +0930] [1410634] [ERROR] Connection in use: ('0.0.0.0', 10086)
[2025-05-29 07:00:07 +0930] [1410634] [ERROR] connection to ('0.0.0.0', 10086) failed: [Errno 98] Address>
[2025-05-29 07:00:08 +0930] [1410634] [ERROR] Connection in use: ('0.0.0.0', 10086)
[2025-05-29 07:00:08 +0930] [1410634] [ERROR] connection to ('0.0.0.0', 10086) failed: [Errno 98] Address>
[2025-05-29 07:00:09 +0930] [1410634] [ERROR] Connection in use: ('0.0.0.0', 10086)
[2025-05-29 07:00:09 +0930] [1410634] [ERROR] connection to ('0.0.0.0', 10086) failed: [Errno 98] Address>
[2025-05-29 07:00:10 +0930] [1410634] [ERROR] Connection in use: ('0.0.0.0', 10086)
[2025-05-29 07:00:10 +0930] [1410634] [ERROR] connection to ('0.0.0.0', 10086) failed: [Errno 98] Address>
[2025-05-29 07:00:11 +0930] [1410634] [ERROR] Can't connect to ('0.0.0.0', 10086)
```

Hopefully this is the same as you found!

--- a/src/dashboard.py
+++ b/src/dashboard.py
@@ -3115,16 +3115,19 @@ def peerInformationBackgroundThread():
     time.sleep(10)
     while True:
         with app.app_context():
-            for c in WireguardConfigurations.values():
-                if c.getStatus():
-                    try:
-                        c.getPeersTransfer()
-                        c.getPeersLatestHandshake()
-                        c.getPeersEndpoint()
-                        c.getPeersList()
-                        c.getRestrictedPeersList()
-                    except Exception as e:
-                        print(f"[WGDashboard] Background Thread #1 Error: {str(e)}", flush=True)
+            try:
+                curKeys = list(WireguardConfigurations.keys())
+                for name in curKeys:
+                    if name in WireguardConfigurations.keys() and WireguardConfigurations.get(name) is not None:
+                        c = WireguardConfigurations.get(name)
+                        if c.getStatus():
+                            c.getPeersTransfer()
+                            c.getPeersLatestHandshake()
+                            c.getPeersEndpoint()
+                            c.getPeersList()
+                            c.getRestrictedPeersList()
+            except Exception as e:
+                print(f"[WGDashboard] Background Thread #1 Error: {str(e)}", flush=True)
         time.sleep(10)
 
 def peerJobScheduleBackgroundThread():

